tests/main/snaps-state/task.yaml

```
@@ -0,0 +1,83 @@
+    snap_path=$("$TESTSTOOLS"/snaps-state pack-local "$SNAP_NAME")
+    snap install --dangerous "${snap_path}"
+    test-snapd-tools.echo test123 | MATCH "test123"
+    snap remove "$SNAP_NAME"
+
+    # Check the local snap file is already created
+    test -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
+    rm -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
+
+    # Try to pack a local snap which does not exist
+    "$TESTSTOOLS"/snaps-state pack-local SNAP_NO_EXIST 2>&1 | MATCH "error: cannot pack"
+
+    # Make and install a snap by using the install-local subcommand
+    snap_path=$("$TESTSTOOLS"/snaps-state install-local "$SNAP_NAME")
+    test-snapd-tools.echo test123 | MATCH "test123"
+    snap remove "$SNAP_NAME"
+
+    # Check the local snap file is already created
+    test -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
+
+    # Make and install a snap when snap file is already created
+    snap_path=$("$TESTSTOOLS"/snaps-state install-local "$SNAP_NAME")
+    test-snapd-tools.echo test123 | MATCH "test123"
+    snap remove "$SNAP_NAME"
+
+    # Check the local snap file is already created
+    test -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
+    rm -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_1.0_all.snap"
+
+    # Make and install a snap by using the install-local-as subcommand
+    snap_path=$("$TESTSTOOLS"/snaps-state install-local-as "$SNAP_NAME" "$SNAP_NAME"_test)
+    test-snapd-tools_test.echo test123 | MATCH "test123"
+    snap remove "$SNAP_NAME"_test
+    rm -f "$TESTSLIB/snaps/${SNAP_NAME}/${SNAP_NAME}_test_1.0_all.snap"
+
+    # Make and install a snap by using the install-local subcommand with --devmode
+    snap_path=$("$TESTSTOOLS"/snaps-state install-local "$SNAP_DEVMODE" --devmode)
+    snap list "$SNAP_DEVMODE"
+    snap remove "$SNAP_DEVMODE"
+    rm -f "$TESTSLIB/snaps/${SNAP_DEVMODE}/${SNAP_DEVMODE}_1.0_all.snap"
+
+    # check the command is-classic-confinement-supported works
+    if [[ "$SPREAD_SYSTEM" == fedora-* ]]; then
+        not "$TESTSTOOLS"/snaps-state is-classic-confinement-supported
+    elif [[ "$SPREAD_SYSTEM" == ubuntu-1* ]]; then
+        "$TESTSTOOLS"/snaps-state is-classic-confinement-supported
+    fi
+
+    # Make and install a snap by using the install-local subcommand with --classic
+    if "$TESTSTOOLS"/snaps-state is-classic-confinement-supported; then
+        snap_path=$("$TESTSTOOLS"/snaps-state install-local "$SNAP_CLASSIC" --classic)
+        snap list "$SNAP_CLASSIC" | MATCH classic$
+        snap remove "$SNAP_CLASSIC"
+        rm -f "$TESTSLIB/snaps/${SNAP_CLASSIC}/${SNAP_CLASSIC}_1.0_all.snap"
+    fi
+
+    # Make and install a snap by using the install-local subcommand with --jailmode
+    if [ "$(snap debug confinement)" = strict ] ; then
+        snap_path=$("$TESTSTOOLS"/snaps-state install-local "$SNAP_JAILMODE" --jailmode)
+        snap list "$SNAP_JAILMODE" | MATCH "jailmode$"
```

Same as `classic$` above:
```suggestion
        snap list "$SNAP_JAILMODE" | MATCH 'jailmode$'
```

---

done